Tests/ApolloTests/DataLoaderTests.swift

```diff
@@ -4,120 +4,79 @@ import ApolloTestSupport
     let loader = DataLoader<Int, Int> { keys in
       batchLoads.append(keys)
-      return Promise(fulfilled: keys)
+      return self.resultsForKeys(keys)
     }
-    
-    let expectation = self.expectation(description: "Waiting for all loads")
-    
+        
     let promises = [loader[1], loader[1]]
+    let values = try promises.map { try $0.get() }
     
-    whenAll(promises).andThen { values in
-      XCTAssertEqual(values, [1, 1])
-      XCTAssertEqual(batchLoads.count, 1)
-      XCTAssertEqual(batchLoads[0], [1])
-      expectation.fulfill()
-    }
-    
-    loader.dispatch()
-    
-    waitForExpectations(timeout: 1)
+    XCTAssertEqual(values, [1, 1])
+    XCTAssertEqual(batchLoads.count, 1)
+    XCTAssertEqual(batchLoads[0], [1])
   }
   
-  func testCachesRepeatedRequests() {
-    var batchLoads: [[String]] = []
+  func testCachesRepeatedRequests() throws {
+    var batchLoads: [Set<String>] = []
     
     let loader = DataLoader<String, String> { keys in
       batchLoads.append(keys)
-      return Promise(fulfilled: keys)
+      return self.resultsForKeys(keys)
     }
-    
-    let expectation1 = self.expectation(description: "Waiting for all loads")
-    
+        
     let promises1 = [loader["A"], loader["B"]]
+    let values1 = try promises1.map { try $0.get() }
     
-    whenAll(promises1).andThen { values in
-      XCTAssertEqual(values, ["A", "B"])
-      XCTAssertEqual(batchLoads.count, 1)
-      XCTAssertEqual(batchLoads[0], ["A", "B"])
-      expectation1.fulfill()
-    }
-    
-    loader.dispatch()
-    
-    waitForExpectations(timeout: 1)
-    
-    let expectation2 = self.expectation(description: "Waiting for all loads")
-    
+    XCTAssertEqual(values1, ["A", "B"])
+    XCTAssertEqual(batchLoads.count, 1)
+    XCTAssertEqualUnordered(batchLoads[0], ["A", "B"])
+        
     let promises2 = [loader["A"], loader["C"]]
+    let values2 = try promises2.map { try $0.get() }
     
-    whenAll(promises2).andThen { values in
-      XCTAssertEqual(values, ["A", "C"])
-      XCTAssertEqual(batchLoads.count, 2)
-      XCTAssertEqual(batchLoads[1], ["C"])
-      expectation2.fulfill()
-    }
-    
-    loader.dispatch()
-    
-    waitForExpectations(timeout: 1)
-    
-    let expectation3 = self.expectation(description: "Waiting for all loads")
+    XCTAssertEqual(values2, ["A", "C"])
+    XCTAssertEqual(batchLoads.count, 2)
+    XCTAssertEqual(batchLoads[1], ["C"])
     
     let promises3 = [loader["A"], loader["B"], loader["C"]]
-    
-    whenAll(promises3).andThen { values in
-      XCTAssertEqual(values, ["A", "B", "C"])
-      XCTAssertEqual(batchLoads.count, 2)
-      expectation3.fulfill()
+    let values3 = try promises3.map { try $0.get() }
+
+    XCTAssertEqual(values3, ["A", "B", "C"])
+    XCTAssertEqual(batchLoads.count, 2)
+  }
+  
+  // - Helpers
+  
+  private func resultsForKeys<Key>(_ keys: Set<Key>) -> [Key: Key] {
+    return keys.reduce(into: [:]) { result, key in
+      result[key] = key
```

Comment:
Yeah, this was how the tests were already written before (by me, what was I thinking). Agree it may be confusing, so maybe I should come up with a simple mapping function instead (like number to letter of the alphabet maybe).

Reply:
Yeah that'd be less confusing, I think.